src/main.rs

```diff
@@ -1,69 +1,64 @@
 // SPDX-FileCopyrightText: 2025 blinry <mail@blinry.org>
+// SPDX-FileCopyrightText: 2025 Joshix <joshix@asozial.org>
 //
 // SPDX-License-Identifier: GPL-3.0-or-later
 
+use procfs::{process::Process, ProcError, ProcResult};
 use std::path::PathBuf;
 
-#[derive(Debug)]
-struct Process {
-    cwd: PathBuf,
-    tty: i32,
+fn crawl_children(pid: i32) -> ProcResult<Vec<(usize, Process)>> {
+    let mut frontier = vec![(0, Process::new(pid)?)];
+    let mut processes: Vec<(usize, Process)> = vec![];
+    while let Some((depth, process)) = frontier.pop() {
+        for task in process.tasks()? {
+            for child in task?.children()? {
```

Comment:
I'm not sure when these fail, but there could be a race condition here. A dying process could cause maybe a failure, resulting in the entire function to fail. That's probably not good.
But I couldn't produce a failure in here, so I don't know.